### PR TITLE
fix: dashboard modal could not cope with deleted tiles

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

@annikaschmid accidentally deleted an insight from a dashboard and then couldn't put it back.

The insight API response was returning the deleted dashboard in the insight's dashboards field. That meant the UI didn't offer the ability to add it to that dashboard because it thought it was already added

Even though there was a test that protected against this which passes and I couldn't figure out why the application behaves well in the test and not when running 🤷 

## Changes

The dashboard_tiles relation is correct. And we should eventually remove the use of the dashboards relation.

So, let's expose it. This begins the deprecation of the dashboards field on the Insight API.

And overwrites the content of the dashboards field with the correct values when returning a representation of the model.

## How did you test this code?

running it locally and seeing it work
updating developer tests

### CODE REVIEW

**Comment by pauldambra** [comment]
Ugh 🤯 

```
Prefetch(
    "dashboards", queryset=Dashboard.objects.filter(id__in=DashboardTile.objects.only("dashboard_id").all())
),
```

This works as expected i.e. if one or more DashboardTiles is deleted then they filter out the dashboard during the prefetch.

Unless **all** of the tiles are deleted. Then `DashboardTile.objects.only("dashboard_id").all()` evaluates to an empty list which is ignored when filtering. So all dashboards are included instead of none. If you then patch back to the API all of the tiles are added back onto the dashboard

😡

**Comment by pauldambra** [comment]
We should highlight this deprecation in the release so we can delete the field without guilt subsequently

**Comment by Twixes** [comment]
Hmm, some tests failing, can you take a look at those? @pauldambra

**Review by Twixes — COMMENTED** [review]
Looks good to me but some checks still red

File: frontend/src/types.ts [hunk 1]
```diff
@@ -1142,7 +1148,9 @@ export interface InsightModel extends Cacheable {
     created_at: string
     created_by: UserBasicType | null
     is_sample: boolean
+    // deprecated: use dashboard_tiles instead
```
Comment (Twixes) [inline frontend/src/types.ts:1151]: For some IDE highlighting magic:
```suggestion
     /** @deprecated Use `dashboard_tiles instead */
```

**Review by Twixes — APPROVED** [review]
:shipit:

**Pull request merged**